src/sdg/api/serializers/producten.py

```diff
@@ -127,24 +168,386 @@ class Meta:
+        publicatie_datum = most_recent_version.pop("publicatie_datum", None)
+        vertalingen = most_recent_version.pop("vertalingen", [])
+
+        validated_data["generiek_product"] = self.get_generiek_product(generiek_product)
+
+        if verantwoordelijke_organisatie:
+            verantwoordelijke_organisatie = self.get_organisatie(
+                verantwoordelijke_organisatie
+            )
+
+        if type(validated_data.get("catalogus")) is dict:
+            validated_data["catalogus"] = self.get_default_catalogus(
+                verantwoordelijke_organisatie
+            )
+
+        if not validated_data.get("catalogus").is_referentie_catalogus:
+            validated_data["referentie_product"] = self.get_referentie_product(
+                validated_data["generiek_product"]
+            )
+
+        if product_valt_onder:
+            if "generiek_product" in product_valt_onder:
+                validated_data["product_valt_onder"] = self.get_product(
+                    product_valt_onder["generiek_product"],
+                    validated_data["catalogus"],
+                )
+
+        if bevoegde_organisatie:
+            if "organisatie" in bevoegde_organisatie:
+                validated_data["bevoegde_organisatie"] = self.get_organisatie(
+                    bevoegde_organisatie["organisatie"]
+                )
+        else:
+            validated_data["bevoegde_organisatie"] = verantwoordelijke_organisatie
+
+        product, created = Product.objects.get_or_create(**validated_data)
+
+        if locaties:
+            product.locaties.set(
+                self.get_locaties(
+                    locaties,
+                    validated_data["catalogus"],
+                )
+            )
+
+        if gerelateerde_producten:
+            gerelateerde_catalogus_producten = []
+            for gerelateerde_product in gerelateerde_producten:
+                if "generiek_product" in gerelateerde_product:
+                    gerelateerde_catalogus_producten.append(
+                        self.get_product(
+                            gerelateerde_product["generiek_product"],
+                            validated_data["catalogus"],
+                        )
+                    )
+            product.gerelateerde_producten.set(gerelateerde_catalogus_producten)
+
+        versie = 1
+
+        if not created:
+            previous_product_versie = ProductVersie.objects.filter(
+                product=product
+            ).first()
+
+            versie = product.most_recent_version.versie
+
+            if previous_product_versie.publicatie_datum:
+                versie += 1
+
+        product_versie, product_versie_created = ProductVersie.objects.update_or_create(
+            product=product,
+            versie=versie,
+            publicatie_datum=None,
+            defaults={"publicatie_datum": publicatie_datum},
+        )
+
+        if vertalingen:
+            for vertaling in vertalingen:
```

Comment:
We always create all available languages, so iterate over TaalChoices, like done here: https://github.com/maykinmedia/sdg-invoervoorziening/pull/654/files#diff-4d8eb0ae426869d19f172e59b7f7d020e47859073f9edae8e288125d53896270R162